frontend/src/pages/AccountSend.vue

```diff
@@ -414,18 +448,55 @@ function useSendForm() {
     if (!token.value?.address) throw new Error(vm.$i18n.tc('Send.select-a-token'));
     if (!recipientId.value) throw new Error(vm.$i18n.tc('Send.enter-a-recipient'));
 
+    sendMax.value = true;
+
     if (NATIVE_TOKEN.value?.address === token.value?.address) {
       if (!userAddress.value || !provider.value) throw new Error(vm.$i18n.tc('Send.wallet-not-connected'));
       const fromAddress = userAddress.value;
       const recipientAddress = await toAddress(recipientId.value, provider.value);
-      const { ethToSend } = await umbraUtils.getEthSweepGasInfo(fromAddress, recipientAddress, provider.value);
-      humanAmount.value = formatUnits(ethToSend, token.value.decimals);
-      return ethToSend;
+      const { ethToSend } = await umbraUtils.getEthSweepGasInfo(fromAddress, recipientAddress, provider.value, {
+        gasLimit: await estimateNativeSendGasLimit(),
+      });
+      humanAmount.value = formatUnits(ethToSend.sub(toll.value), token.value.decimals);
+    } else {
+      const tokenBalance = balances.value[token.value.address];
+      humanAmount.value = formatUnits(tokenBalance.toString(), token.value.decimals);
+    }
+  }
+
+  // Get an accurate estimate of the amount of gas needed to perform a native send.
+  async function estimateNativeSendGasLimit() {
+    // Increase estimate to give us some wiggle room if network conditions are volatile.
+    let scaleFactor;
+    switch (chainId.value) {
+      case 137:
+        scaleFactor = '120';
+        break;
+      default:
+        scaleFactor = '110';
     }
 
-    const tokenBalance = balances.value[token.value.address];
-    humanAmount.value = formatUnits(tokenBalance.toString(), token.value.decimals);
-    return tokenBalance.toString();
+    return (
+      await umbra.value!.umbraContract.estimateGas.sendEth(
+        // We will be sending to an address that has never been seen before, which increases gas
+        // costs by 25k. To ensure this cost is included in our gas limit estimate, we estimate
+        // using a `to` address that is randomly generated (and thus likely to have never been seen
+        // before).
+        Wallet.createRandom().address,
```

Comment:
We can avoid introducing dependency on the wallet class with ` new RandomNumber().asHex.replace(/0/g, 'f').replace(/^./, '0').slice(0, 42)`